humanoid_league_team_communication/scripts/TeamCommTest.py

```diff
@@ -12,11 +12,12 @@
 if __name__ == '__main__':
     rospy.init_node("TeamCommTest")
     ball_pub = rospy.Publisher("balls_relative", PoseWithCertaintyArray, queue_size=1)
-    goal_post_pub = rospy.Publisher("goal_post_relative", PoseWithCertaintyArray, queue_size=1)
-    position_pub = rospy.Publisher("position", PoseWithCovarianceStamped, queue_size=1)
-    obstacle_pub = rospy.Publisher("obstacle_relative", ObstacleRelativeArray, queue_size=1)
-    position_msg = PoseWithCovarianceStamped()
+    goal_post_pub = rospy.Publisher("goal_posts_relative", PoseWithCertaintyArray, queue_size=1)
+    position_pub = rospy.Publisher("pose_with_certainty", PoseWithCertainty, queue_size=1)
+    obstacle_pub = rospy.Publisher("obstacles_relative", ObstacleRelativeArray, queue_size=1)
```

Comment:
The obstacles, balls, etc are relative to our `base_footprint`. We probably want to transform them into the `map` coordinate system. Is there any definition regarding this in the protocol definition?

Reply:
the convention mentioned in the protocol is the following:
x meters along the field with 0 at the centre of the field and positive towards opponent goals
y meters across the field with 0 at the centre of the field and positive to the left
z meters above the field with 0 at the surface of the field and positive up
(see [https://github.com/RoboCup-Humanoid-TC/RobocupProtocol/blob/master/robocup_extension.proto](https://github.com/RoboCup-Humanoid-TC/RobocupProtocol/blob/master/robocup_extension.proto))

So the data should be transformed as you said, so other teams can use it in a mixed match. However, this could be done in the TeamComm internally, so this would not affect the rest of the software stack if it is transformed back before publishing the information.

Reply:
Is this transformation implemented yet?

Reply:
No

Reply:
Okay no hurry, I am only interested, because I haven't done anything with the Teamkom.